cleverhans/attacks.py

```diff
@@ -486,6 +486,102 @@ def parse_params(self, eps=2.0, num_iterations=1, xi=1e-6, clip_min=None,
         return True
 
 
+class CarliniWagnerL2(Attack):
```

Comment:
This is simply a suggestion: most other attacks use a name that reflects at the high-level how they work. On the other hand, the attack is referred to as Carlini and Wagner in the literature. So I'm not sure what naming convention is most appropriate for here.